src/update-process-env.js

```diff
@@ -33,9 +40,23 @@ export default function updateProcessEnv (launchEnv) {
   }
 }
 
-function getEnvFromShell () {
+function shellShouldBePatched () {
   let shell = process.env.SHELL
-  if (shell && (shell.endsWith('/bash') || shell.endsWith('/sh'))) {
+  if (!shell) {
+    return false
+  }
+  for (let s of OSX_SHELLS_TO_PATCH) {
+    if (shell.endsWith(s)) {
+      return shell
+    }
+  }
+
+  return false
+}
```

Comment:
I would suggest two fairly minor stylistic changes to this `shellShouldBePatched` function:

1. I think it should *take* a shell as an argument and return a boolean, rather than internally getting the shell from `process.env` and returning either a string or `null`.

2. I think the name is a bit odd; it makes it sound like the shell itself is getting patched. Maybe `canGetLoginEnvFromShell(shellPath)`?

I think this would make the function a little simpler to test, since you wouldn't have to overwrite `process.env.SHELL`. You wouldn't need a loop; you could just do:

```js
expect(canGetLoginEnvFromShell('/bin/bash')).toBe(true)
expect(canGetLoginEnvFromShell('/some/other/bash')).toBe(true)
expect(canGetLoginEnvFromShell('/usr/local/bin/fish')).toBe(true)

expect(canGetLoginEnvFromShell('/bin/give-me-cash')).toBe(false)
expect(canGetLoginEnvFromShell('/usr/bin/birthday-bash')).toBe(false)
```